pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -33,8 +33,12 @@ export class FileAutoComplete extends React.Component {
             displayFiles: [],
             isOpen: false,
             value: this.props.value || null,
+            type: this.props.type
         };
-        this.updateFiles(props.value || '/');
+        if (this.props.type === "file") {
+            this.props.onChange(this.props.value.slice(this.props.value.lastIndexOf("/") + 1));
+        }
+        this.updateFiles(props.value ? this.props.type === "file" ? props.value.slice(0, props.value.lastIndexOf("/") + 1) : props.value : "/");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19108-20230718-131821-c41f31c8-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)